Minutes Technical Lead Call 3 December2024 (#200)
ESMValGroup/Community: Tech Lead/Minutes/20241203.md

- Absentees: none
- Chair: @valeriupredoi
- Minutes taker: @valeriupredoi

## Action Items

| Meeting | Assignees | Item | Status |
|-|-|-|-|
| 2024-12-03 | @rbeucher | Arrange meeting on Intake | Pending |
| 2024-12-03 | @ehogan   | Split docs meeting | Pending |
| 2024-08-13 | all       | Review [RTW pull request](https://github.com/ESMValGroup/ESMValTool/pull/3210)|Done, thank you everyone! :tada: |

## Agenda and topics discussed

- Review of Pending Action Items
- @ehogan: From last time: A new Geoscience RSE community has recently been formed (https://society-rse.org/weather-and-climate-rsecon24-birds-of-a-feather-session-retrospective-blog-post/) and we are interested in organising a townhall meeting for RSEs at EGU25 next year; will there be an ESMValTool presence there and should we get involved?
- @ehogan: I have created a RTW project board (https://github.com/orgs/ESMValGroup/projects/27/views/1) to help @sloosvel and I (others are welcome!) work together to get RTW working for v2.12.0 release :tada: @bettina-gier: where to put smaller test recipes, Emma says tag her - still to be decided;

> @bouweandela: test ones on CircleCI - but data must be small, since they get downloaded/run every night, see [CI test configuration](https://github.com/ESMValGroup/ESMValTool/blob/ba9cfe426f9f2a97dba672784ccba8462eea7d6e/.circleci/config.yml#L177-L231) -> these test Core mostly;

Reviewer: ```suggestion
@bouweandela: [testing](https://github.com/ESMValGroup/ESMValTool/tree/main/esmvaltool/recipes/testing) ones are run on CircleCI (see [ESMValGroup/ESMValCore#1636](https://github.com/ESMValGroup/ESMValCore/issues/1636) for some background information) - but data must be small, since they get downloaded/run every night, see [CI test configuration](https://github.com/ESMValGroup/ESMValTool/blob/ba9cfe426f9f2a97dba672784ccba8462eea7d6e/.circleci/config.yml#L177-L231) -> these test Core mostly;
```

> - @ehogan: Chris is back working on ESMValTool until he finishes with his Apprenticeship (~May 2025). Is there anything, perhaps something non-critical-path-Dask-related, that he could get involved in? - @bouweandela: could run all recipes with the `dask.Distributed` scheduler, not all work, it's a case-by-case challenging work, @ehogan proposes using the simple scheduler first; @bouweandela also proposes `pyproject.toml`-ing ESMValTool; @schunma is very keen on Dask optimization with `Distributed`, @bouweandela proposes Manu work mostly on it, with Chris testing;

Reviewer: ```suggestion
- @ehogan: Chris is back working on ESMValTool until he finishes with his Apprenticeship (~May 2025). Is there anything, perhaps something non-critical-path-Dask-related, that he could get involved in? - @bouweandela: could run all recipes with the `distributed` scheduler, not all work, it's a case-by-case challenging work, @ehogan proposes using the Recipe Test Workflow first; @bouweandela also proposes `pyproject.toml`-ing ESMValTool; @schunma is very keen on making the Dask configuration more flexible [ESMValGroup/ESMValCore#2371](https://github.com/ESMValGroup/ESMValCore/issues/2371), @bouweandela proposes Manu work mostly on it, with Chris testing;
```

> - @ehogan: Using the new Software Carpentry template [ESMValTool Tutorial Issue #335](https://github.com/ESMValGroup/ESMValTool_Tutorial/issues/335)

Reviewer: ```suggestion
- @ehogan: Made progress on using the new Software Carpentry template [ESMValTool Tutorial Issue #335](https://github.com/ESMValGroup/ESMValTool_Tutorial/issues/335) with external help, but it is challenging to do this ourselves. @bouweandela recommends merging all outstanding pull requests in the tutorial repository ASAP and switch to the new template while help is available.
```

> - @ehogan: Should we arrange a meeting to come up with a documentation design following this [comment on "Agenda for ESMValTool community workshop: 19-21 November"](https://github.com/ESMValGroup/Community/discussions/192#discussioncomment-11258741) - @rbeucher is keen to get Felicity onboard; Emma will organize the first meeting on this via a Doodle, this will be a **to-do item** here;

Reviewer: ```suggestion
- @ehogan: Should we arrange a meeting to come up with a documentation design following this [comment on "Agenda for ESMValTool community workshop: 19-21 November"](https://github.com/ESMValGroup/Community/discussions/192#discussioncomment-11258741) - @rbeucher is keen to get Felicity onboard; Emma will organize the first meeting on this via a Doodle, this will be an **action item** here;
```

- @ehogan: Somehow at the MO we have ended up with multiple date ranges for HadISST:
  ```
  % ls /data/users/esmval/ESMValTool/temporary/obs/Tier2/HadISST/ 
  OBS_HadISST_reanaly_1_Amon_ts_187001-201712.nc
  OBS_HadISST_reanaly_1_Amon_ts_187001-202112.nc
  OBS_HadISST_reanaly_1_OImon_sic_187001-201712.nc
  OBS_HadISST_reanaly_1_OImon_sic_187001-202112.nc
  OBS_HadISST_reanaly_1_Omon_tos_187001-201712.nc
  OBS_HadISST_reanaly_1_Omon_tos_187001-202112.nc
  ```

>   I found [ESMValTool: PR #3072: Update HadISST cmorizer with recent years](https://github.com/ESMValGroup/ESMValTool/pull/3027), so I understand why the date ranges changed, but is it expected that ESMValTool will error with this obs setup? Temporary solution: delete older datasets, keep latest dataset with latest version;

Reviewer: ```suggestion
  I found [ESMValTool: PR #3072: Update HadISST cmorizer with recent years](https://github.com/ESMValGroup/ESMValTool/pull/3027), so I understand why the date ranges changed, but is it expected that ESMValTool will error with this obs setup? Temporary solution: delete older datasets, keep latest dataset with latest version; @bouweandela This is the result of lacking support for versioning in OBS datasets. I have recommended following the same versioning scheme as for OBS4MIPs, but this seems more work than the community has time for at the moment. See [ESMValGroup/ESMValTool#3452](https://github.com/ESMValGroup/ESMValTool/issues/3452) and [ESMValGroup/ESMValCore#1943](https://github.com/ESMValGroup/ESMValCore/pull/1943) for more background information.
```

- @ehogan: Slack for more general use Ask the UET to join, possibly wider community to join as well;

> - @rbeucher: intake discussion - @charles-turner-1 started looking at adding datasets; a meeting would be useful with Bouwe, Tina, the Met Office contractor might have switched focus, if at all. @rbeucher will organize a meeting, **this is a to-do item**

Reviewer: ```suggestion
- @rbeucher: intake discussion - @charles-turner-1 started looking at adding datasets; a meeting would be useful with Bouwe, Tina, the Met Office contractor might have switched focus, if at all. @rbeucher will organize a meeting, **this is an action item**
```
